mlflow/utils/environment.py

```diff
@@ -729,6 +731,122 @@ def _get_pip_install_mlflow():
         return f"pip install mlflow=={VERSION} 1>&2"
 
 
+def _add_requirements_to_file(
+    new_reqs: List[Requirement],
+    file_path: str,
+) -> None:
+    """
+    Adds new requirements to the given requirements file (either `conda.yaml`
+    or `requirements.txt`). If any of the new requirements overlap with the
+    requirements in the file, the new requirements will overwrite the old ones.
+
+    Args:
+        new_reqs (List[`Requirement`]): List of new requirements to add to the file.
+        file_path (str): Path to the locally-saved requirements file. Must point to
+            either a `conda.yaml` or `requirements.txt` file.
+    """
+    old_reqs = _get_requirements_from_file(file_path)
+    updated_reqs = _add_or_overwrite_requirements(new_reqs, old_reqs)
+    _write_requirements_to_file(file_path, updated_reqs)
+    _logger.info(
+        "Done adding requirements!\n\n"
+        f"Old requirements:\n{pformat([str(req) for req in old_reqs])}\n\n"
+        f"Updated requirements:\n{pformat(updated_reqs)}\n"
+    )
+
+
+def _remove_requirements_from_file(
+    reqs_to_remove: List[Requirement],
+    file_path: str,
+) -> None:
+    """
+    Removes requirements from the given requirements file (either `conda.yaml`
+    or `requirements.txt`). If any of the provided requirements are not present
+    in the file, they will be ignored. This function also ignores version specifiers.
+    For example, if the file contains "pandas==1.2.3", and the provided requirement
+    is "pandas", the function will remove "pandas==1.2.3" from the file.
+
+    Args:
+        reqs_to_remove (List[`Requirement`]): List of requirements to remove from the file
+        file_path (str): Path to the locally-saved requirements file. Must point to
+            either a `conda.yaml` or `requirements.txt` file.
+    """
+    old_reqs = _get_requirements_from_file(file_path)
+    updated_reqs = _remove_requirements(reqs_to_remove, old_reqs)
+    _write_requirements_to_file(file_path, updated_reqs)
+    _logger.info(
+        "Done removing requirements!\n\n"
+        f"Old requirements:\n{pformat([str(req) for req in old_reqs])}\n\n"
+        f"Updated requirements:\n{pformat(updated_reqs)}\n"
+    )
+
+
+def _get_requirements_from_file(
```

Comment:
We might want to simplify this (it's a private function, after all) by not worrying about the file names (that's all internal configuration that isn't mutable by users) and to standardize with pathlib:

```python
def _get_requirements_from_file(
    file_path: str,
) -> List[Requirement]:
    
    path = pathlib.Path(file_path)
    data = path.read_text()
    
    if path.name == _CONDA_ENV_FILE_NAME:
        extracted = yaml.safe_load(data)
        data = _get_pip_deps(extracted)
    else:
        data = data.splitlines()
    
    return [Requirement(req) for req in data]
```

Reply:
nice! pathlib seems a lot more convenient, and agreed that we probably don't need to care about the `else` filenames